src/Microsoft.DotNet.ProjectModel/ProjectReader.cs

```diff
@@ -246,6 +243,38 @@ public Project ReadProject(Stream stream, string projectName, string projectPath
             return project;
         }
 
+        internal static string NormalizeProjectDirectoryPath(string path)
+        {
+            if (File.Exists(path) &&
+                string.Equals(Path.GetFileName(path), Project.FileName, StringComparison.OrdinalIgnoreCase))
+            {
+                string directoryName = Path.GetDirectoryName(path);
+                if (string.IsNullOrEmpty(directoryName))
+                {
+                    directoryName = Directory.GetCurrentDirectory();
+                }
+
+                return Path.GetFullPath(directoryName);
+            }
+            else if (Directory.Exists(path) &&
+                     File.Exists(Path.Combine(path, Project.FileName)))
+            {
+                return Path.GetFullPath(path);
+            }
+
+            return null;
```

Comment:
Not really.  The general rule I've used in the past is 'null' and 'empty string' should be handled the same way.  That's why there is a "IsNullOrEmpty" method. So one is just as good as the other.

Also - this is what the existing code did.